CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+# Overview
+
+This documents explains the processes and practices recommended for
+contributing enhancements to the Prometheus charm.
+
+- Generally, before developing enhancements to this charm, you should consider
+  [opening an issue ](https://github.com/canonical/prometheus-operator) explaining
+  your the use case.
+- If you would like to chat with us about your use-cases or proposed
+  implementation, you can reach us at Canonical
+  [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev).
+- It is strongly recommended that prior to engaging in any enhancements
+  to this charm you familiarise your self with the Juju
+- Familiarising yourself with the
+  [Juju Operator SDK](https://juju.is/docs/sdk) and the Python
+  [Operator Framework](https://ops.readthedocs.io/en/latest/)
+  library will help you a lot when working on PRs.
+- All enhancements require review before being merged. Besides the
+  code quality and test coverage, the review will also take into
+  account the resulting user experience for Juju administrators using
+  this charm. Please help us out in having easier reviews by rebasing
+  onto the main branch, avoid merge commits and enjoy a linear Git
+  history.
+
+## Developing
+
+Create and activate a virtualenv with the development requirements:
+
+```bash
+$ virtualenv -p python3 venv
+$ source venv/bin/activate
+$ pip install -r requirements-dev.txt
+```
+
+### Setup
+
+A typical setup using [snaps](https://snapcraft.io/), for deployments
+to a [microk8s](https://microk8s.io/) cluster can be done using the
+following commands
+
+```bash
+$ sudo snap install microk8s --classic
+$ microk8s.enable dns
```

Comment:
You either need to include the commands to add the user to the microk8s group, or use `sudo` here. Personally, I wouldn't include these steps, but rather link to the development setup doc at https://juju.is/docs/sdk/dev-setup and just document the requirement to create the storage pool after that's complete. (You can also omit the install of charmcraft below if you point at the dev setup page)

Reply:
OK. Changed as suggested.